text/069-coq-roadmap.md

```diff
@@ -89,6 +89,12 @@ to add and remove items, to reflect the evolution of the project.
 
 #### Rewrite rules
 
+The goal is to add (unsafe) user-defined rewrite rules. This features allows users to add computation rules to axioms which can be useful for prototyping. It also allows for different kinds of computation rules with respect to what Coq currently permits: non-linearity, overlapping left-hand sides (*eg* one can write an addition on natural numbers that reduces on both sides: `0 + n` and `n + 0` both reducing to `n`).
```

Comment:
"prototyping" isn't mentioned in the CEP (#50).  It suggests that there are later steps to make something "final" or "production".  Perhaps you could explain what you have in mind in the CEP with a brief explanation of those later steps?

What fraction of our users will be willing/able to use this feature for important projects if it's unsafe?

I assume the prototyping is meant for users and not just Coq developers.

Reply:
I don't understand why it should be mentioned in the CEP. These are just examples of what people want to do with rewrite rules: in particular we already have people using the fork of @yannl35133 to try out various type theories that are not otherwise encodable in Coq.

This use case has nothing to do with the status of rewrite rules.

It is true however, that the first instance we propose is something that is unsafe—in the sense that Coq may not terminate, or be incomplete for conversion if the user did not choose the right rewrite rules—and that the user must know that. To me this is the same as unsetting guard checking or using `Type : Type`, these are features used by actual people in actual projects even though they are not *safe*.

Reply:
It's that "prototyping" is unclear.  As I said:

> It suggests that there are later steps to make something "final" or "production".

What are those steps?  Or is it a misleading choice of word?  Can you clarify?

Reply:
I think there is a misunderstanding. A user may want to use rewrite rules for prototyping before they actually implement something, but it's up to the user so *I* don't have to explain the steps for this hypothetical user work.